app/controllers/hello_react_controller.rb

```diff
@@ -1,2 +1,137 @@
+    insert_cupe
+    insert_signoff
+  end
+
+  private
+  def insert_title
+    grid([0,0], [1,0]).bounding_box do
+      stroke_color 100, 100, 0, 0
+      text "Logo will go here"
+      rectangle [-35, 180], 300, 100
+      stroke
+    end
+
+    grid([1,0], [2,0]).bounding_box do
+      font "Times-Roman"
+      font_size 20
+      text_box "TEACHING ASSISTANT CONTRACT", style: :bold, at: [0, 130], align: :center
+      text_box "#{@session} Session #{@year}", style: :bold, at: [0, 110], align: :center
+    end
+  end
+
+  private
+  def insert_intro
+    grid([2,0],[3,0]).bounding_box do
+      font_size 12
+      text "Dear #{@first_name} #{@last_name},"
+      move_down 5
+      text "We are please to offer you a position as a Teaching Assistant in the Department of Computer Science " +
+      "during the period of <b>#{@start_date}</b> to <b>#{@end_date}</b>. Your assignment will be <b>#{@position}</b>. (We reserver the " +
+      "right to change your appointment from this course to another course, if necessary). The total time involved " +
+      "in the assignment is <b>#{@hours}</b>. The salary for this position is governed by the collective aggreement " +
+      "between the University of Toronto and the Canadian Union of Public Employees, Local 3902, a copy of which " +
+      "will be provided upon request. The current rate of pay for UG = $43.65; SGS I and SGS II/PDF = $43.65. " +
+      "According to the collective aggreement the total sum paid to you for the appintment described above will be :",
+      indent_paragraphs: 30, inline_format: true
+    end
+  end
+
+  private
+  def insert_calculations
+    grid([4,0],[5,0]).bounding_box do
+      stroke_color 100, 100, 0, 0
+      rectangle [100, 130], 300, 100
+      text_box "calculations go here", at: [120, 110]
+      stroke
+    end
+  end
+
+  private
+  def insert_cupe
+    grid([6,0],[8,0]).bounding_box do
+      text "If this is your first appointment  after  becoming  a  Ph.D.  student,  you  will  be  guaranteed  three  more " +
+      "appointments of the same number of hours in subsequent years as per Article 16.05 of the CUPE 3902 (Unit1) Collective Agreement." +
+      "If this is not your first appointment, you may still be owed one or more subsequent " +
+      "appointments under the aforementioned Article. If this is the case, you may consult the letter you received last April, where your " +
+      "appointment status was summarized. Please feel free to contact Karen Reid to seehow various guarantees fit together.",
+      indent_paragraphs: 30
+
+      move_down 10
+
+      text "You will soon be given the opportunity to review the Description of Duties and Allocation of Hours(DDAH) form, which will set out " +
+      "more specifically the duties of your position, and the hours assigned to each.",
+      indent_paragraphs: 30
+
+      text "Please also be advised that in accepting this	offer	of appointment, you are accepting the responsibility " +
```

Comment:
It seems like there is a lot of excess spacing between words in this paragraph, and in a few cases in the paragraph above?